dotnet/Gherkin/CucumberMessages/Pickles/PickleCompiler.cs

```diff
@@ -168,27 +168,24 @@ protected virtual void CompileScenarioOutline(List<Pickle> pickles,
     protected virtual PickleStep CreatePickleStep(Step step, string text, PickleStepArgument argument, IEnumerable<string> astNodeIds)
     {
         CurrentStepKeywordType = GetKeywordType(step, CurrentStepKeywordType);
-        return new PickleStep(argument, astNodeIds, idGenerator.GetNewId(), text, CurrentStepKeywordType);
+        return new PickleStep(argument, astNodeIds.ToList(), idGenerator.GetNewId(), CurrentStepKeywordType, text);
     }
 
-    protected virtual StepKeywordType GetKeywordType(Step step, StepKeywordType lastStepKeywordType)
+    protected virtual PickleStepType GetKeywordType(Step step, PickleStepType lastStepKeywordType)
     {
         var stepKeywordType = step.KeywordType;
-        switch (stepKeywordType)
+        return stepKeywordType switch
         {
-            case StepKeywordType.Context:
-            case StepKeywordType.Action:
-            case StepKeywordType.Outcome:
-            case StepKeywordType.Unknown:
-                return stepKeywordType;
-            case StepKeywordType.Conjunction:
-                return lastStepKeywordType == StepKeywordType.Unspecified ||
-                       lastStepKeywordType == StepKeywordType.Unknown
-                    ? StepKeywordType.Context
-                    : lastStepKeywordType;
-            default:
-                return StepKeywordType.Unspecified;
-        }
+            Io.Cucumber.Messages.Types.StepKeywordType.CONTEXT => PickleStepType.CONTEXT,
+            Io.Cucumber.Messages.Types.StepKeywordType.ACTION => PickleStepType.ACTION,
+            Io.Cucumber.Messages.Types.StepKeywordType.OUTCOME => PickleStepType.OUTCOME,
+            Io.Cucumber.Messages.Types.StepKeywordType.UNKNOWN => PickleStepType.UNKNOWN,
+            Io.Cucumber.Messages.Types.StepKeywordType.CONJUNCTION =>
+                (lastStepKeywordType == PickleStepType.UNKNOWN
+                                ? PickleStepType.CONTEXT
```

Comment:
Note to self: This switch appears to be correctly migrated. But also different from (at least) the java implementation. 

A scenario with the `*` keyword would be all `Context` steps in .Net, but `Unknown` in Java.

```
 * Step A
 * Step b
 * Step c
```

Reply:
@clrudolphi this diverges from other implementations. Can be fixed after this is merged. One thing at a time. :wink: 

https://github.com/cucumber/gherkin/pull/427